mautic/mautic
# [DPMMA-2234] Country statistics for campaigns/emails (table)

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



This PR adds email/contacts stats to tables in email and campaign views and allows data export. It aligns with the values shown on the map in  https://github.com/mautic/mautic/pull/12570

Email campaign (/s/campaigns/view/{campaign_id})
![table_campaign_all](https://github.com/mautic/mautic/assets/102536220/6dc04b22-2d47-47bb-9605-065876e87060)

Campaign without sending emails (/s/campaigns/view/{campaign_id})
![table_campaign_no_emails](https://github.com/mautic/mautic/assets/102536220/6ff4b2fe-01fc-42a5-9dab-9317fe9defaf)

Email (s/emails/view/{email_id})
![table_email](https://github.com/mautic/mautic/assets/102536220/2e4c0e72-df17-4ea3-a52d-3a28bbe37b2e)


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create at least one email. Send it to several contacts (preferably with a non-empty country field), then open and click on the link of at least one of messages.
3. Email statistics should be correctly displayed in the e-mail view (s/emails/view/{email_id}) "Countries" tab.
4. Create and run a test campaign with at least one "email.send" event.
5. Simulate reads and clicks on messages sent from the campaign.
6. Statistics from all emails sent from the campaign should be displayed in campaign preview (/s/campaigns/view/{campaign_id}) in "Countries" tab.
